services/issue/label.go

```diff
@@ -59,12 +65,12 @@ func RemoveLabel(issue *models.Issue, doer *user_model.User, label *models.Label
 		return models.ErrRepoLabelNotExist{}
 	}
 
-	if err := models.DeleteIssueLabel(issue, label, doer); err != nil {
+	if err := models.DeleteIssueLabel(ctx, issue, label, doer); err != nil {
 		return err
 	}
 
 	notification.NotifyIssueChangeLabels(doer, issue, nil, []*models.Label{label})
-	return nil
+	return committer.Commit()
```

Comment:
There are a lot of database operations inside `NotifyIssueChangeLabels`.

It's not ideal to call `NotifyIssueChangeLabels` without transaction, it would cause deadlocks.

Since the code has been changed to this, I wouldn't against, just point out it's not ideal.

Reply:
A workaround is to use `WithContext`, and put `NotifyIssueChangeLabels` outside the transaction.

Reply:
Yes, commit should run before notification.